### DISCUSSION TITLE
Rockwell Controllers Drivers

### DISCUSSION THREAD

**Discussion by AlexCastro-Prog** [frangoteam/FUXA#199] [Ideas]
Hello FrangoTeam,

The Fuxa Web Scada is awesome. I was wonder if you could add the Library and drivers for Rockwell Automation controllers like Pycomm3 so in this way all the PLC controllers will be covered.

**Comment by unocelli** [comment]
Hi, thanks for the appreciation.
we have discussed the issue in #118 and #72. The library that we can try to use is [node-ethernet-ip](https://github.com/cmseaton42/node-ethernet-ip) and this should work. My problem is that I don't use rockwell controllers and I should be able to test it, e.g. a simulator (as simple as possible) that I don't have.

**Comment by AlexCastro-Prog** [comment]
Hi Unocelli, I have on my office all rockwell PLC like PLC 5, SLC 500, Micrologix, Micro 800, Compactlogix and Contrologix, so feel free if you want through thee VPN connection use my plc's to add this node. I work 99% with Rockwell PLC's and I would like to see Fuxa Web Scada working with rockwell PLC. Is there something that I can do for you to enable Fux to talk with Rockwell PLC's?

**Comment by AlexCastro-Prog** [comment]
I'm very interesting to make FUXA Web SCADA work with Rockwell plc. I can help you with my PLCs so, in this way, we can do a test together.

**Comment by AlexCastro-Prog** [comment]
Also, I have the software for Rockwell and everything related to Rockwell Automation.

**Comment by AlexCastro-Prog** [comment]
For the last one, I found this node [https://github.com/plcpeople/nodepccc](url) that We can try to add to FUXA because this node can talk with all Controllers for Rockwell Automation. Let me know if I can help you to add this node to FUXA. I'm very excited to help you.

**Reply by jorger1953** [reply]
> For the last one, I found this node [https://github.com/plcpeople/nodepccc](url) that We can try to add to FUXA because this node can talk with all Controllers for Rockwell Automation. Let me know if I can help you to add this node to FUXA. I'm very excited to help you.

Alex, with this driver we have to map all the tags that we wnat to read, on PLC5/SLC format por Controllogiz and Compactlogix, stopping the PLC. Why do yo suggest to use this?

**Comment by unocelli** [comment]
looks promising, have you already used this library (nodepccc) ?

**Reply by AlexCastro-Prog** [reply]
Yes, I have been tested and it works great!!!!!! If you can add this library over there your SCADA will be powerful because FUXA will be able to communicate with Siemens and Rockwell. Amazing!!!

**Reply by unocelli** [reply]
Ok, give me some time to finish a task and then I want to try it. I will announce to you as soon as possible to make tests and to help me ;)

**Reply by AlexCastro-Prog** [reply]
Perfect Count on me! Let's test together! Let me know when you are free to start the tests. Thank you!

**Comment by unocelli** [comment]
Hi,
the working is in progress in the [ethernetip ](https://github.com/frangoteam/FUXA/tree/ethernetip) branch, maybe a connection can be made (install the plugins first).
I suppose we need to specify the {routing: [0x01,0x00,0x01,0x00]}? have you a idea how we can configure it in the GUI?

**Reply by AlexCastro-Prog** [reply]
Hello There,

I don't have any IDEA. let me see how I do on Node-RED and I'll let you know

**Comment by AlexCastro-Prog** [comment]
Hello unoclli,

I tested yesterday and it worked great. nodepccc is tested and working on Nod-RED the sam node that you will put on Fuxa Webscada.

Can we schedule a virtual meeting to help you to do a GUI for that? What do you think? My email address is axandre1@hotmail.com I can share my screen with you with ControlLogix, Compactlogix, Micrologix working ith this node. Let me know...

**Comment by AlexCastro-Prog** [comment]
Hello Frango Team!

I made a video that helps your users to connect rockwell plc's to FUXA web scada.

Please take a look and live your comment. Please!

Click on link below:

https://youtu.be/iTz7ryccLWw

**Reply by onght** [reply]
Hi Alex, I installed the latest version of Fuxa but I am not able to see EthernetIP as a Device type in the dropdown list. What do I need to do to get it listed in the selection list? Thanks.

**Reply by onght** [reply]
I found what I missed. I need to manually install the EthernetIP device in the Plugin setting.

**Reply by csacomani** [reply]
Dear Onght, how can I do this manual Ethernet/IP installation?

**Reply by onght** [reply]
Go to the Editor screen --> Setup --> Connection and add a device. 

Give a meaningful name to the device and select the EthernetIP type. Set the IP address and for EIP, the port should be 44818.

Let me know if you still have problem
